accountsdb-plugin-postgres/src/accountsdb_plugin_postgres.rs

```diff
@@ -0,0 +1,349 @@
+                    msg: format!(
+                        "The config file is not in the JSON format expected: {:?}",
+                        err
+                    ),
+                })
+            }
+            Ok(config) => {
+                let connection_str = format!("host={} user={}", config.host, config.user);
+                match Client::connect(&connection_str, NoTls) {
+                    Err(err) => {
+                        return Err(AccountsDbPluginError::Custom(
+                            Box::new(AccountsDbPluginPostgresError::DataStoreConnectionError {
+                                msg: format!(
+                                "Error in connecting to the PostgreSQL database: {:?} host: {:?} user: {:?} config: {:?}",
+                                err, config.host, config.user, connection_str),
+                            })));
+                    }
+                    Ok(mut client) => {
+                        let result = client.prepare("INSERT INTO account (pubkey, slot, owner, lamports, executable, rent_epoch, data, updated_on) \
+                            VALUES ($1, $2, $3, $4, $5, $6, $7, $8) \
+                            ON CONFLICT (pubkey) DO UPDATE SET slot=$2, owner=$3, lamports=$4, executable=$5, rent_epoch=$6, \
+                            data=$7, updated_on=$8");
+
+                        match result {
+                            Err(err) => {
+                                return Err(AccountsDbPluginError::Custom(
+                                    Box::new(AccountsDbPluginPostgresError::DataSchemaError {
+                                    msg: format!(
+                                        "Error in preparing for the accounts update PostgreSQL database: {:?} host: {:?} user: {:?} config: {:?}",
+                                        err, config.host, config.user, connection_str
+                                    ),
+                                })));
+                            }
+                            Ok(update_account_stmt) => {
+                                self.client = Some(Mutex::new(PostgresSqlClientWrapper {
+                                    client,
+                                    update_account_stmt,
+                                }));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Unload all plugins and loaded plugin libraries, making sure to fire
+    /// their `on_plugin_unload()` methods so they can do any necessary cleanup.
+    fn on_unload(&mut self) {
+        info!("Unloading plugin: {:?}", self.name());
+    }
+
+    fn update_account(&mut self, account: ReplicaAccountInfoVersions, slot: u64) -> Result<()> {
+        match account {
+            ReplicaAccountInfoVersions::V0_0_1(account) => {
+                if let Some(accounts_selector) = &self.accounts_selector {
```

Comment:
Why empty `accounts_selector` from config means select nothing? It looks strange.